### PR TITLE
[CI][BugFix] Fix failure CI "amd-v1-sample-plus-logits-mi300-1"

### PR DESCRIPTION
**Pull request by SoluMilken**
## Purpose

Fix failure CI: `amd-v1-sample-plus-logits-mi300-1`
https://buildkite.com/vllm/ci/builds/65147/canvas?jid=019e06e7-5882-42b3-8c44-6723201294ea&tab=output
https://buildkite.com/vllm/ci/builds/65109/canvas?jid=019e0656-76cd-445e-8324-cfe4b042f924&tab=output
https://buildkite.com/vllm/ci/builds/65102/canvas?jid=019e0637-d718-43e0-b285-eaab513fd44e&tab=output

## Root Cause

- The test relied on fork semantics: monkeypatched entry_points was expected to be visible in workers.
- ROCm CI uses spawn instead of fork, so spawned workers do not inherit the in-memory monkeypatch.
- As a result, the dummy logits processor entrypoint was not discovered in EngineCore/worker.
- The model generated normal tokens instead of the forced target token sequence.
- Recent subprocess failure propagation made the existing issue visible.

## Test Plan

```bash
python -m pytest \
"tests/v1/logits_processors/test_custom_offline.py::test_custom_logitsprocs[CustomLogitprocSource.LOGITPROC_SOURCE_ENTRYPOINT]" \
"tests/v1/logits_processors/test_custom_online.py::test_custom_logitsprocs[server0-facebook/opt-125m]" \
-v \
-p no:warnings
```

## Test Result

image

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://docs.vllm.ai/en/latest/contributing>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: tests/v1/logits_processors/utils.py [hunk 1]
```diff
@@ -189,3 +192,26 @@ def new_req_logits_processor(
...
+        f"[{LOGITSPROCS_GROUP}]\n"
+        f"{DUMMY_LOGITPROC_ENTRYPOINT} = {DUMMY_LOGITPROC_FQCN}\n",
+        encoding="utf-8",
+    )
+
+    monkeypatch.syspath_prepend(str(tmp_path))
+    monkeypatch.setenv(
+        "PYTHONPATH",
+        str(tmp_path) + os.pathsep + os.environ.get("PYTHONPATH", ""),
+    )
```
[the remark this answers is not in the document]
Comment (SoluMilken) [reply tests/v1/logits_processors/utils.py:218]: [287b48e](https://github.com/vllm-project/vllm/pull/42106/commits/287b48e68f13acbe33b5e1c43331f4eace8a156c)

**Comment by SoluMilken** [comment]
Hi @afeldman-nm @dzhengAP could you please take a look?

This fixes the ROCm logits processor entrypoint test failures by replacing the in-process `importlib.metadata.entry_points` monkeypatch with a temporary real `.dist-info/entry_points.txt`. The old test relied on fork semantics, but ROCm runs these paths under spawn, so EngineCore/worker processes did not inherit the monkeypatch and failed to discover the dummy logits processor.

Thanks.

**Comment by dzhengAP** [comment]
@SoluMilken thanks!there is an earlier PR discussed to fix this fork issue. Are they similar? https://github.com/vllm-project/vllm/pull/42040

**Comment by SoluMilken** [comment]
Hi [dzhengAP](https://github.com/dzhengAP), they are almost the same. Let me close this PR.
